From 007f850049034a41ca0cc7aae808efd5e2a51941 Mon Sep 17 00:00:00 2001
From: Clayton Craft <clayton@craftyguy.net>
Date: Sat, 25 Oct 2025 14:06:44 -0700
Subject: [PATCH] udev: also handle change actions

In some cases the kernel seem to change LED/backlight devices and
permissions get reset back to root:root, so this re-applies the desired
permissions when those changes happen.
---
 90-brightnessctl.rules | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/90-brightnessctl.rules b/90-brightnessctl.rules
index 99a4c4d..8940471 100644
--- a/90-brightnessctl.rules
+++ b/90-brightnessctl.rules
@@ -1,4 +1,4 @@
-ACTION=="add", SUBSYSTEM=="backlight", RUN+="/bin/chgrp video /sys/class/backlight/%k/brightness"
-ACTION=="add", SUBSYSTEM=="backlight", RUN+="/bin/chmod g+w /sys/class/backlight/%k/brightness"
-ACTION=="add", SUBSYSTEM=="leds", RUN+="/bin/chgrp input /sys/class/leds/%k/brightness"
-ACTION=="add", SUBSYSTEM=="leds", RUN+="/bin/chmod g+w /sys/class/leds/%k/brightness"
+ACTION=="add|change", SUBSYSTEM=="backlight", RUN+="/bin/chgrp video /sys/class/backlight/%k/brightness"
+ACTION=="add|change", SUBSYSTEM=="backlight", RUN+="/bin/chmod g+w /sys/class/backlight/%k/brightness"
+ACTION=="add|change", SUBSYSTEM=="leds", RUN+="/bin/chgrp input /sys/class/leds/%k/brightness"
+ACTION=="add|change", SUBSYSTEM=="leds", RUN+="/bin/chmod g+w /sys/class/leds/%k/brightness"
-- 
2.51.1

